### ISSUE TITLE
builder.build() causes error

### ISSUE THREAD

**Issue by lcucurachi** [davemckelvie/things-drivers#17]
I am trying to use this on my raspberry pi3 but the following appers:

```
FATAL EXCEPTION: main
                                                                       Process: com.example.thingspoc, PID: 11497
                                                                       java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/things/pio/PeripheralManager;
                                                                           at nz.geek.android.things.drivers.i2c.BaseI2cDevice.getBus(BaseI2cDevice.java:35)
                                                                           at nz.geek.android.things.drivers.i2c.Pcf8574.create(Pcf8574.java:62)
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd.createPort(I2cSerialCharLcd.java:101)
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd.<init>(I2cSerialCharLcd.java:91)
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd.<init>(I2cSerialCharLcd.java:24)
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd$I2cSerialCharLcdBuilder.build(I2cSerialCharLcd.java:305)
                                                                           at com.example.thingspoc.HomeActivity.onCreate(HomeActivity.java:48)
                                                                           at android.app.Activity.performCreate(Activity.java:7000)
                                                                           at android.app.Activity.performCreate(Activity.java:6991)
                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                           at android.os.Looper.loop(Looper.java:164)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                        Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.things.pio.PeripheralManager" on path: DexPathList[[zip file "/system/framework/com.google.android.things.jar", zip file "/data/app/com.example.thingspoc-v9qqJdW3eyVBjOk27aslVQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.thingspoc-v9qqJdW3eyVBjOk27aslVQ==/lib/arm, /system/lib, /vendor/lib]]
                                                                           at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
                                                                           at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
                                                                           at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                           at nz.geek.android.things.drivers.i2c.BaseI2cDevice.getBus(BaseI2cDevice.java:35) 
                                                                           at nz.geek.android.things.drivers.i2c.Pcf8574.create(Pcf8574.java:62) 
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd.createPort(I2cSerialCharLcd.java:101) 
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd.<init>(I2cSerialCharLcd.java:91) 
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd.<init>(I2cSerialCharLcd.java:24) 
                                                                           at nz.geek.android.things.drivers.lcd.I2cSerialCharLcd$I2cSerialCharLcdBuilder.build(I2cSerialCharLcd.java:305) 
                                                                           at com.example.thingspoc.HomeActivity.onCreate(HomeActivity.java:48) 
                                                                           at android.app.Activity.performCreate(Activity.java:7000) 
                                                                           at android.app.Activity.performCreate(Activity.java:6991) 
                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214) 
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731) 
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856) 
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                           at android.os.Looper.loop(Looper.java:164) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
```

**Comment by davemckelvie** [comment]
Android Things developer preview 0.7 introduced some breaking changes to the API. You need to update your device to use developer preview 0,7, or use an earlier version of this library (1.3.0).

This library has been updated to work with Android Things developer preview 0.7.

See here for further details: https://stackoverflow.com/questions/49141634/peripheralmanagerservice-throws-noclassdeffounderror

**Comment by LoKKeR96** [comment]
Thank you. I have updated my raspberry pi and I'm using now the libraries version 1.7. The only thing is that it's still not working. There is no error now being thrown but the 16x2 LCD just doesn't display anything. I've connected it via LCM1602 and I'm using an level shifter and I'm using my oscilloscope to check whether the raspberry pi is outputting anything but it's not... What could be wrong? I have no idea now.
